src/main/proto/th2_grpc_data_provider/data_provider.proto

```diff
@@ -28,86 +28,69 @@ option java_package = "com.exactpro.th2.dataprovider.grpc";
 service DataProvider {
 
   /* returns a single event with the specified id */
-  rpc getEvent (EventID) returns (EventData) {
+  rpc getEvent (EventID) returns (EventResponse) {
   }
 
   /* returns a list of events with the specified ids */
-  rpc getEvents (EventIds) returns (Events) {
+  rpc getEvents (BulkEventRequest) returns (BulkEventResponse) {
   }
 
   /* returns a single message with the specified id */
-  rpc getMessage (MessageID) returns (MessageData) {
+  rpc getMessage (MessageID) returns (MessageGroupResponse) {
   }
 
   /* returns a list of message stream names */
-  rpc getMessageStreams (MessageStreamNamesRequest) returns (StringList) {
+  rpc getMessageStreams (MessageStreamsRequest) returns (MessageStreamsResponse) {
   }
 
-  /* Creates a message stream that matches the filter. */
-
-  rpc searchMessages (MessageSearchRequest) returns (stream StreamResponse) {
+  /* creates a message stream that matches the filter. */
+  rpc searchMessages (MessageSearchRequest) returns (stream MessageSearchResponse) {

```

Comment:
some options of MessageSearchRequest can exclude/ignore others. Maybe split it into several methods without this disadvantage